internal/extension/manager.go

```diff
@@ -141,19 +143,34 @@ func newExtensionService(dag *nilGuardedPointer[StateAwareDAG]) extpb.ExtensionS
 }
 
 func (s *extensionService) Subscribe(_ *emptypb.Empty, stream grpc.ServerStreamingServer[extpb.Event]) error {
+	channel := BlockingDAG.newUpdateChannel()
 	for {
-		time.Sleep(time.Second * 5)
-		if err := stream.Send(&extpb.Event{}); err != nil {
-			return err
+		dag := <-channel
+
+		opts := []cel.EnvOption{
+			kuadrant.CelExt(&dag),
+		}
+		if env, err := cel.NewEnv(opts...); err == nil {
+			s.subscribtions.Range(func(sub, _ interface{}) bool {
+				if prg, err := env.Program(sub.(subscription).cAst); err == nil {
+					if _, _, err := prg.Eval(sub.(subscription).input); err == nil {
+						_ = stream.Send(&extpb.Event{})
```

Comment:
I could have some input on this @adam-cattermole 
How do we want to go about this? Thinking pushing any `error` down to the extension might actually be desirable here, wdyt?